store/generated_key/v2/eigenda.go

```diff
@@ -77,23 +80,40 @@ func NewStore(
 		rbnRecencyWindowSize: rbnRecencyWindowSize,
 		disperser:            disperser,
 		retrievers:           retrievers,
-		verifier:             verifier,
+		certVerifier:         certVerifier,
 	}, nil
 }
 
 // Get fetches a blob from DA using certificate fields and verifies blob
 // against commitment to ensure data is valid and non-tampered.
-func (e Store) Get(ctx context.Context, key []byte) ([]byte, error) {
-	var cert coretypes.EigenDACert
-	err := rlp.DecodeBytes(key, &cert)
-	if err != nil {
-		return nil, fmt.Errorf("RLP decoding EigenDA v2 cert: %w", err)
+func (e Store) Get(ctx context.Context, versionedCert certs.VersionedCert) ([]byte, error) {
+	var cert coretypes.RetrievableEigenDACert
+
+	switch versionedCert.Version {
+	case certs.V0VersionByte, certs.V1VersionByte:
+		var v2Cert coretypes.EigenDACertV2
+		err := rlp.DecodeBytes(versionedCert.SerializedCert, &v2Cert)
+		if err != nil {
+			return nil, fmt.Errorf("RLP decoding EigenDA v2 cert: %w", err)
+		}
+		cert = &v2Cert
+
+	case certs.V2VersionByte:
+		var v3Cert coretypes.EigenDACertV3
+		err := rlp.DecodeBytes(versionedCert.SerializedCert, &v3Cert)
+		if err != nil {
+			return nil, fmt.Errorf("RLP decoding EigenDA v3 cert: %w", err)
+		}
+		cert = &v3Cert
+
+	default:
```

Comment:
this is nice! Think it should be turned into a function `parseCert` or something, which returns the sum-type interface that `GetPayload` accepts. It should prob also be in core such that we can use it elsewhere (for example in some debugging library where people with proxy errors can just input their commitment and itll tell them which cert version it is from).

Prob can move/refactor this once we move proxy to core repo though? I had this ticket for adding such a function: https://linear.app/eigenlabs/issue/DAINT-465/certrbn-parsing-golang-library

Reply:
sounds like a plan!

Reply:
The plan being for me to refactor this after we move proxy to monorepo? If so I'll update my ticket